storage/store.go

```diff
@@ -1243,30 +1239,20 @@ func (s *Store) ExecuteCmd(ctx context.Context, call proto.Call) error {
 		}
 		return r.Next()
 	}
+	var rng *Range
+	var err error
+	// Add the command to the range for execution; exit retry loop on success.
 	for r := retry.Start(s.ctx.RangeRetryOptions); next(&r); {
-		// Add the command to the range for execution; exit retry loop on success.
-		reply.Reset()
-
 		// Get range and add command to the range for execution.
-		rng, err := s.GetRange(header.RaftID)
-		if err != nil {
-			reply.Header().SetGoError(err)
-			return err
-		}
-
-		addedReply, err := rng.AddCmd(ctx, args)
-		if addedReply != nil {
-			gogoproto.Merge(reply, addedReply)
-		}
-		if reply.Header().Error != nil {
-			panic(proto.ErrorUnexpectedlySet)
-		}
+		rng, err = s.GetRange(header.RaftID)
 		if err != nil {
-			reply.Header().SetGoError(err)
+			return nil, err
 		}
 
+		var reply proto.Response
+		reply, err = rng.AddCmd(ctx, args)
```

Comment:
Shadowing would've been fine here if we were using the old closure-based retry interface instead of the loop-based one. I feel like the loop-based retry interface increased the amount of subtle interaction between scopes and at-a-distance error handling.